packages/x-components/src/components/base-grid.vue

```diff
@@ -201,45 +208,45 @@
       );
 
       /**
-       * Checks if a given value is an `ElementRef` object.
-       *
-       * @param value - The value to check.
-       * @returns `true` if the value is an `ElementRef` object, `false` otherwise.
+       * Updates the number of columns rendered inside the grid.
        *
        * @internal
        */
-      const isElementRef = (value: any): value is ElementRef => {
-        return value && value.$el instanceof HTMLElement;
+      const updateRenderedColumnsNumber = (): void => {
+        if (rootEl.value) {
+          const { gridTemplateColumns } = getComputedStyle(rootEl.value);
+          renderedColumnsNumber.value = gridTemplateColumns.split(' ').length;
+        }
       };
 
       /**
-       * Updates the number of columns rendered inside the grid.
+       * Initialises the rendered columns number and sets a ResizeObserver to keep it updated.
        *
        * @internal
        */
-      const updateRenderedColumnsNumber = (): void => {
-        const { gridTemplateColumns } = getComputedStyle(
-          isElementRef(gridEl.value) ? gridEl.value.$el : (gridEl.value as Element)
+      onMounted(() => {
+        /**
+         * Using querySelector to get the root element of the grid instead of using the ref directly
+         * because with the introductions of Fragments, Vue returns every DOM element inside the component.
+         */
+        rootEl.value = document.querySelector('.x-base-grid');
```

Comment:
Related to https://github.com/vuejs/core/issues/4830 and https://github.com/vuejs/core/issues/4830